uqcsbot/scripts/whatsdue.py

```diff
@@ -0,0 +1,68 @@
+from datetime import datetime
+from uqcsbot import bot, Command
+from uqcsbot.scripts.uq_course_util import (get_course_assessment,
+                                            HttpException,
+                                            CourseNotFoundException,
+                                            ProfileNotFoundException)
+
+# Maximum number of courses supported by !whatsdue to reduce call abuse.
+COURSE_LIMIT = 6
+
+def get_formatted_assessment_item(assessment_item):
+    '''
+    Returns the given assessment item in a pretty message format to display to
+    a user.
+    '''
+    course, task, due, weight = assessment_item
+    return f'*{course}*: `{weight}` _{task}_ *({due})*'
+
+@bot.on_command('whatsdue')
+def handle_whatsdue(command: Command):
+    '''
+    `!whatsdue [-f] [--full] [COURSE CODE 1] [COURSE CODE 2] ...` - Returns all
+    the assessment for a given list of course codes that are scheduled to occur
+    after today. If unspecified, will attempt to return the assessment for the
+    channel that the command was called from. If -f/--full is provided, will
+    return the full assessment list without filtering by cutoff dates.
+    '''
+    channel = command.channel
+    command_args = command.arg.split() if command.has_arg() else []
+
+    is_full_output = False
+    if '--full' in command_args:
+        command_args.remove('--full')
+        is_full_output = True
+    if '-f' in command_args:
+        command_args.remove('-f')
+        is_full_output = True
+
+    # If we have any command args left, they're course names. If we don't,
+    # attempt to instead use the channel name as the course name.
+    course_names = command_args if len(command_args) > 0 else [channel.name]
+
+    if len(course_names) > COURSE_LIMIT:
+        bot.post_message(channel, f'Cannot process more than {COURSE_LIMIT} courses.')
+        return
+
+    # If no full output specified, set the cutoff to today's date. Else, set the
+    # cutoff to UNIX epoch (i.e. filter nothing).
+    cutoff_datetime = datetime.fromtimestamp(0) if is_full_output else datetime.today()
+    try:
+        assessment = get_course_assessment(course_names, cutoff_datetime)
+    except HttpException as e:
+        bot.post_message(channel, f'An error occurred, please try again.')
+        return
+    except CourseNotFoundException as e:
+        bot.post_message(channel, f'Could not find course `{e.course_name}`.')
```

Comment:
If we make my suggested change from earlier we can turn this into `bot.post_message(channel, e.message)` (or `str(e)` if you decide to do that - I'm not so sure on that one being the right call). Can then just reraise for logging purposes.